Adds how to export atlas using PyQGIS

SrNetoChan opened the pull request:
### Description
<!---
Include a few sentences describing the overall goals for this Pull Request.
--->
Goal:

<!---
If your PR fixes a ticket, add `fix` in front of the ticket number. The ticket will be closed automatically.
Add as much as needed `fix #number` if the PR closes more than one ticket.
If your PR doesn't fix entirely the ticket, just add the ticket reference.
The complete list of the issues is at https://github.com/qgis/QGIS-Documentation/issues.
-->
Ticket: #

<!---
Indicate whether the fix should be backported to previous release.
Replace the space between square brackets by a `x` to make it checked.
-->
- [x] Backport to LTR documentation is required

### Minimal requirements for merging *(for maintainers)*
<!---
Reviewing is a process done by community members, mostly on a volunteer basis.
We try to keep the overhead as small as possible and appreciate if you help us.
In order for your PR to get merged it should satisfy some minimal requirements:
--->

- [ ] The description of this PR is coherent with the manual and does not provide wrong information.
- [ ] This PR passes the checks. <!---The results will be reported by travis-ci **after** opening the PR.-->

<!---
Please read carefully our writing guidelines at https://docs.qgis.org/testing/en/docs/documentation_guidelines/index.html
to help you fulfill these requirements. Feel free to ask in a comment if you have troubles with any of them.
--->

nyalldawson commented on source/docs/pyqgis_developer_cookbook/composer.rst:
```diff
@@ -206,6 +206,18 @@ To export a layout, the :class:`QgsLayoutExporter <qgis.core.QgsLayoutExporter>`
+
+.. code-block:: python
+
+   exporter.exportToPdf(layout.atlas(), pdf_path, 'pdf', QgsLayoutExporter.PdfExportSettings())
```
pdf_path is actually a "base path" - it will be modified by any atlas setting for filename by expression. So e.g. a path of "c:/my stuff/output" will cause pdfs to be generated with names like "c:/my stuff/output_1.pdf", "c:/my stuff/output_2.pdf", etc. (or "c:/my stuff/output.pdf" if the atlas is set to a single-file pdf output!). But that's a lot of detail to try to fit in here...

nyalldawson commented:
Some other comments on this page which might be nice to include

- layout.initializeDefaults() # this initializes the layout with some default settings, specifically by adding an empty A4 page to the layout. You can create layouts without calling this method, but you'll need to take care of adding pages to the layout yourself
- "this item tells the libraries where to put the map itself. Here we create a map and stretch it over the whole paper size" - the code doesn't do the stretch part. I think it'll create a 0 width/0 height item by default!. You probably want something like `map.attemptMove(QgsLayoutPoint(0,0, QgsUnitTypes.LayoutMillimeters))` and `map.attemptResize(QgsLayoutSize(200,200, QgsUnitTypes.LayoutMillimeters))`. Note that these methods are called "attempt..." because certain layout or item settings (e.g. data defined position or size) will override the call - so you are asking nicely, but aren't necessarily guaranteed that the item WILL move or resize to what you've asked!
- layout.addItem( ... ) - this should be layout.addLayoutItem( ... )

havatv commented on source/docs/pyqgis_developer_cookbook/composer.rst:
```diff
@@ -206,6 +206,18 @@ To export a layout, the :class:`QgsLayoutExporter <qgis.core.QgsLayoutExporter>`
+
+If you want to export all pages from a layout that has the atlas option
+configured and enabled, you need to use the :meth:`atlas()
+<qgis.core.QgsLayout.atlas>` in the exporter :class:`QgsLayoutExporter
```
```suggestion
<qgis.core.QgsLayout.atlas>` method in the exporter (:class:`QgsLayoutExporter
```

SrNetoChan replied:
Done :-)

SrNetoChan commented:
I have changed the previous example a bit to get a base_path variable, and have added a comment about how the exporter uses the filename expression to write the outputs. Hoping that it's more clear like that.

Moved @nyalldawson comments/improvements to a separated issue #3976 so we can address it later.
